src/task/RegisteredActivityTask.ts

```diff
@@ -0,0 +1,30 @@
+import * as types from "durable-functions";
+import { AtomicTask } from "./AtomicTask";
+import { RetryOptions } from "../RetryOptions";
+import { CallActivityAction } from "../actions/CallActivityAction";
+import { CallActivityWithRetryAction } from "../actions/CallActivityWithRetryAction";
+import { RetryableTask } from "./RetryableTask";
+
+export class RegisteredActivityTask extends AtomicTask implements types.RegisteredActivityTask {
+    withRetry: (retryOptions: RetryOptions) => RetryableTask;
+
+    constructor(activityName: string, input?: unknown) {
+        super(false, new CallActivityAction(activityName, input));
+
+        this.withRetry = (retryOptions: RetryOptions): RetryableTask => {
+            if (this.alreadyScheduled) {
+                throw new Error(
+                    "This activity has already been scheduled. You cannot call withRetry on an already scheduled task. Create a new task instead."
```

Comment:
nitpicky nit -> How about something like this:

```suggestion
                    "Invalid use of `.withRetry`: attempted to add retry options to an already scheduled task. The affected task has ID={this.taskScheduledId} and name={this.activityName}. Make sure you only call `.withRetry` on tasks that have not previously been yielded.
```

Note, I don't think this suggestion compiles, but I think you get the idea of what I'm trying to do
(1) Summarize the error type: "Invalid use of .withRetry"
(2) Provide enough identifying information for the customer to find the culprit line of code: printing taskScheduledId and the activityName
(3) The fix: only call withRetry on unyielded tasks.

Reply:
If you're good with this error message format, please also apply it to the orchestration case.

Reply:
Sounds good to me. I reworded it a bit and added the right variables, but I think I got the gist of it. Let me know if the new error message works